server/src/test/java/io/crate/planner/optimizer/rule/EliminateCrossJoinTest.java

```diff
@@ -430,5 +434,71 @@ public void test_resolve_multiple_aliases_with_same_name_in_logical_join_plan()
         var graph = JoinGraph.create(thirdJoin, Function.identity());
         assertThat(graph.size()).isEqualTo(4);
         assertThat(graph.edges()).hasSize(4);
+        assertThat(graph.nodes().get(0)).isEqualTo(rename_a);
+        assertThat(graph.nodes().get(1)).isEqualTo(b);
+        assertThat(graph.nodes().get(2)).isEqualTo(rename_c);
+        assertThat(graph.nodes().get(3)).isEqualTo(d);
+    }
+
+    @Test
+    public void test_graph_with_order() throws Exception {
+        var order = new Order(a, new OrderBy(List.of(x)));
+        Symbol firstJoinCondition = e.asSymbol("a.x = b.y");
+        var firstJoin = new JoinPlan(order, b, JoinType.INNER, firstJoinCondition);
+        Symbol secondJoinCondition = e.asSymbol("b.y = c.z");
+        var join = new JoinPlan(firstJoin, c, JoinType.INNER, secondJoinCondition);
+
+        assertThat(join).isEqualTo(
```

Comment:
Have you considered using `hasOperators()` instead?
minor: if you stick to isEqual() you can maybe beautify it by using `"""` blocks instead.